linalg_ops/attention/generate_e2e_attention_tests.py

```diff
@@ -190,16 +231,29 @@ def generate_function(
     )
 
     query_shape, key_shape, value_shape, result_shape = get_tensor_shapes(shapes_scale)
+    query_shape0 = int_or_question_mark(query_shape[0])
+    query_shape1 = int_or_question_mark(query_shape[1])
+    query_shape2 = int_or_question_mark(query_shape[2])
+    key_shape0 = int_or_question_mark(key_shape[0])
+    key_shape1 = int_or_question_mark(key_shape[1])
+    key_shape2 = int_or_question_mark(key_shape[2])
+    value_shape0 = int_or_question_mark(value_shape[0])
+    value_shape1 = int_or_question_mark(value_shape[1])
+    value_shape2 = int_or_question_mark(value_shape[2])
+    result_shape0 = int_or_question_mark(result_shape[0])
+    result_shape1 = int_or_question_mark(result_shape[1])
+    result_shape2 = int_or_question_mark(result_shape[2])
+
     query_tensor_type = (
-        f"tensor<{query_shape[0]}x{query_shape[1]}x{query_shape[2]}x{query_type.value}>"
+        f"tensor<{query_shape0}x{query_shape1}x{query_shape2}x{query_type.value}>"
     )
```

Comment:
I'm trying to think of a way to refactor this so it isn't so verbose... but this is at least simple and easy to read, so it's probably fine.

Reply:
I can make it shorter, but it will affect its readability.